log-viewer/modules/TreeView.ts

```diff
@@ -492,6 +502,65 @@ function hideByDuration(elements: Array<HTMLElement>) {
   }
 }
 
+// Find the parent node of "elm" (or return null if we reach the root of the tree)
+function getParentNode(elm: Element | null) {
+  if (elm === null || elm.classList.contains('root')) {
+    return null;
+  }
+
+  let parent = elm.parentElement;
+  while (parent && !parent.classList.contains('node')) {
+    parent = parent.parentElement;
+  }
+
+  return parent;
+}
+
+function insertCrumb(container: Element, node: HTMLElement) {
+  // @ts-ignore (custom dom property)
+  const line = node.line as LogLine,
+    nameNode = node.querySelector<HTMLElement>('.name') || node,
+    crumb = document.createElement('div'),
+    textElm = document.createElement('div'),
+    textNode = document.createTextNode(line.getBreadcrumbText());
+
+  crumb.classList.add('crumb');
+  crumb.title = nameNode.textContent!;
+  crumb.appendChild(textElm);
+  crumb.addEventListener('click', evt => {
```

Comment:
I know we will not have loads of these but I think the event would be better handled on the breadcrumb parent element?

Reply:
I had a look at it but it doesn't seem worth the extra effort of identifying the breadcrumb (when we probably clicked on a child element) and then finding the node to show...